src/preview/PreviewView.tsx

```diff
@@ -1,7 +1,56 @@
+import { Header } from "common/components/header/Header"
+import {
+    Text,
+    Box,
+    Flex,
+    Grid,
+    GridItem,
+  } from "@chakra-ui/react";
+import { useNavigate, useParams } from "react-router";
+import { FaHome } from "react-icons/fa";
+import { useGetExercisePublication } from "./hooks";
+import { Sections } from "./components/Sections";
+import { SkeletonScreen } from "common/components/SkeletonScreen";
+import { RenderSection } from "./components/RenderSection";
+
 export const PreviewView: React.FC = () => {
+    const navigate = useNavigate();
+    const handleNavigateHome = () => {
+        navigate(`/`);
+    }
+
+    const {id: exerciseId} = useParams<string>();
+    const {data: exercisePublication, isLoading} = useGetExercisePublication(exerciseId!);
+    const publicationSections = exercisePublication?.publicationSections || [];
+
     return (
-        <div>
-            Preview Page :D
-        </div>
+        isLoading ? <SkeletonScreen /> : 
+        <>
+            <Box mt={"20px"} mr={"20px"} mx={"3%"}>
+                <Header>
+                    <Flex width="100%" align="center" justifyContent="center">
+                        <Box as={FaHome} size="25px" cursor="pointer" onClick={handleNavigateHome} ml={[2, 4, 6]} />
+                        <Text fontSize="2xl" fontWeight="normal" mx="auto" textAlign="center">
+                            {exercisePublication?.publicationTitle}
+                        </Text>
+                    </Flex>
+                </Header>
+                <Grid templateColumns={"30% 70%"} mt={"3%"}>
+                    <GridItem>
+                        <Sections publicationSections={publicationSections}/>
+                    </GridItem>
+                    <GridItem
+                        css={{
+                            maxHeight: "calc(100vh - 140px)",
+                            overflowY: "auto",
+                        }}
+                    >
+                        {publicationSections.map((publicationSection, index) => (
+                            <RenderSection publicationSection={publicationSection} sectionIndex={String(index)} key={index}/>
```

Comment:
There was another requirement, when user scrolls on this component, we should update the selectedSectionIndex as well so I think you will have to lift the state up and handle this requirement as well.

Reply:
That functionality is taken care in selectAndScrollToSection function in Sections component. When user clicks on a particular section the section is highlighted and user is scrolled to that section's content.